purchases/src/main/kotlin/com/revenuecat/purchases/PostReceiptHelper.kt

```diff
@@ -145,10 +147,21 @@ internal class PostReceiptHelper(
         storeUserID: String?,
         marketplace: String?,
         initiationSource: PostReceiptInitiationSource,
+        purchaseState: PurchaseState,
         onSuccess: (PostReceiptResponse) -> Unit,
         onError: PostReceiptDataErrorCallback,
     ) {
         val presentedPaywall = paywallPresentedCache.getAndRemovePresentedEvent()
+
+        if (purchaseState == PurchaseState.PENDING) {
```

Comment:
This check has been moved from the previous position in the `PostTransactionWithProductDetailsHelper`. Plan is to have pending transactions get to this point. And before returning an error here, cache all transaction metadata, unifying where we cache all this transaction metadata into a single place.